fees/dependencies.go

```diff
@@ -0,0 +1,11 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fees
+
+type Metadata interface {
```

Comment:
This isn't a dependency of `fees`, it is what `fees` implements. 

You should define this in `chain`.